src/compiler/crystal/codegen/debug.cr

```diff
@@ -29,13 +36,16 @@ module Crystal
     end
 
     def fun_metadatas
-      @fun_metadatas ||= {} of LLVM::Function => LibLLVMExt::Metadata
+      @fun_metadatas ||= {} of LLVM::Function => Array(Tuple(String, LibLLVMExt::Metadata))
```

Comment:
Let's not use `Tuple` here. Later on there are a bunch of `[0][1]` and `[0][0]` which are very hard to understand.

Let's introduce a new type:

```crystal
record FunMetadata, filename : String, metadata : LibLLVMExt::Metadata
```

And then:

```crystal
@fun_metadatas ||= {} of LLVM::Function => Array(FunMetadata)
```

Once you do that the compiler will guide you to fix the relevant code.

Reply:
Good point. Will refactor.